File: python/02_Factorial.py
'''
	계승(Factorial) 구하기
		1) 재귀적 용법
		2) 비재귀적 용법
'''

'''
# Factorial: 반복적 용법
def Factorial(num) :
    res = 1
    for i in range(1, num+1) :
        res = res * i
    return  res
'''

# Factorial: 재귀적 용법
def Factorial(num) :
    if num==0 :		# 재귀 함수 탈출(종료) 조건
        return 1
    return num * Factorial(num-1) 

if __name__ == "__main__" :
    num = int(input('임의의 정수: '))
    print(f'Factorial {num}: {Factorial(num)}')

File: c/02_Factorial.c
#include <stdio.h>
int  Factorial(int num); 
int main(void)	
{
	int	num;

	printf("임의의 정수: ");
	scanf_s("%d", &num);
  	// scanf("%d", &num);
	
  	printf("%d Factorial: %d \n", num, Factorial(num) );
	return 0;
}

# Factorial: 재귀적 용법
int  Factorial(int num)	{
	if(num == 0)
		return 1;	
	return num * Factorial (num - 1);
}

/*
# Factorial: 반복적 용법
int  Factorial(int num)	{
	int	res = 1;
	for (int i = 1; i <= num; i++)
		res  =  res  *  i;
	return  res;
}
*/
